sw/device/lib/testing/mask_rom_test.h

```diff
@@ -0,0 +1,93 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_TESTING_MASK_ROM_TEST_H_
+#define OPENTITAN_SW_DEVICE_LIB_TESTING_MASK_ROM_TEST_H_
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace mask_rom_test {
+
+/**
+ * Mixin for global mocks.
+ *
+ * `GlobalMock<T>` is a subclass of `T` that can have at most one instance at a
+ * time (checked at runtime) and makes this instance globally accessible via the
+ * static `GlobalMock<T> &Instance()` method.
+ *
+ * Mock classes used in tests should be of this type so that mock functions can
+ * access their instances during tests. Since we prefer using strict mocks,
+ * mock classes should also be of type `testing::StrictMock`. Mock classes that
+ * satisfy both requirements can be defined using a type alias as follows:
+ *
+ *     namespace mask_rom_test {
+ *     namespace internal {
+ *     class MockFoo {
+ *       ...
+ *     };
+ *     }  // namespace internal
+ *     // Type alias for making `internal::MockFoo` a global and strict mock.
+ *     using MockFoo = GlobalMock<testing::StrictMock<internal::MockFoo>>;
+ *     ...
+ *     }  // namespace mask_rom_test
+ */
+template <typename T>
+class GlobalMock : public T {
```

Comment:
Honestly, I feel like CTRP is nicer here, but I understand why you'd do this, to be consistent with some of the googletest "mixins".